app/src/main/java/org/oppia/android/app/profile/PinPasswordActivityPresenter.kt

```diff
@@ -257,4 +269,38 @@ class PinPasswordActivityPresenter @Inject constructor(
         dialog.dismiss()
       }.create().show()
   }
+
+  private fun ensureProfileIsOnboarded() {
+    val profile = checkNotNull(pinViewModel.profile.value)
+
+    if (!profile.completedProfileOnboarding) {
+      launchOnboardingScreen(profile.id, profile.name)
+    } else {
+      launchHomeScreen()
+    }
+  }
+
+  private fun launchOnboardingScreen(profileId: ProfileId, profileName: String) {
+    val introActivityParams = IntroActivityParams.newBuilder()
+      .setProfileNickname(profileName)
+      .setParentScreen(IntroActivityParams.ParentScreen.PIN_PASSWORD_SCREEN)
+      .build()
+
+    val intent = IntroActivity.createIntroActivity(activity)
```

Comment:
This seems a bit odd to me--shouldn't `createIntroActivity` be properly setting up its intent?

Reply:
Yes - I have done a refactor so that we are passing the profileId and arguments to the Intent helper instead.